enhancements/installer/feature-sets.md

```diff
@@ -0,0 +1,209 @@
+---
+title: installer-feature-sets
+authors:
+  - @patrickdillon
+reviewers:
+  - @wking, installer, upgrades
+  - @bparees, feature sets
+approvers:
+  - @sdodson
+  - @zaneb
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - @deads2k
+creation-date: 2022-09-07
+last-updated: 2022-09-07
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/CORS-2253
+---
+
+# Installer Feature Sets
+
+## Summary
+
+This enhancement proposes adding support for feature sets to `openshift-install`.
+Users would be able to enable feature sets at install time. The Installer would support all feature sets
+in the OpenShift API, as well as an additional `ExperimentalNoUpgrade` feature set, which would be used
+to gate Installer features that are in active development.
+
+## Motivation
+
+Feature gates have been standard in OpenShift since at least 4.1, but have been absent from the Installer.
+There are many observable benefits of the practice, but the primary motivations are:
+
+* Align the Installer with OpenShift release practices and graduation criteria, i.e. enable QE & CI testing
+of gated features
+* Improve Installer developer experience by removing concerns about exposing non-GA features during active development
+* Allow a mechanism for pre-release user testing
+
+We have an immediate use case for this as we are targeted to release a tech preview feature in 4.12.
+
+### User Stories
+
+As a member of OpenShift concerned with the release process (TRT, dev, staff engineer, maybe even PM), I want to opt in to
+pre-release features so that I can run periodic testing in CI and obtain a signal of feature quality.  
+
+As quality engineering, I want to opt in to pre-release features so that I can test nightly and other release images 
+both in CI & locally.
+
+As a customer, I want to opt in to pre-release features so that I can test and verify proof of concepts. 
+
+As an Installer developer, I want to protect new API fields with a feature gate so that I can build features without concern
+that an incomplete feature would be exposed to customers.
+
+### Goals
+
+The goals of this proposal are:
+* To provide an API that will allow users to enable feature sets in the Installer
+* Allow the Installer to enable feature sets in the cluster
+* Ensure the proposed implementation adheres to best practices for feature sets
+
+### Non-Goals
+
+None
+
+## Proposal
+
+### Workflow Description
+
+Example workflow demonstrating feature set validation and use:
+
+**Developer** contributes code to the openshift-install codebase.
+
+**Admin** is the user performing an install.
+
+1. Developer adds feature `foo` to openshift-install. Feature `foo`
+includes field `bar` in the install config; feature `foo` is part of the Tech Preview Feature Set.
```

Comment:
I'm struggling to remember the order that things happen in. Does it go
1. create install config
2. render all manifests using the installer
3. install the cluster using the manifests rendered in item 2
?

Reply:
Yes, although step 2 can be either explicit or implicit.